samza-core/src/test/java/org/apache/samza/zk/TestZkJobCoordinator.java

```diff
@@ -214,47 +243,53 @@ public void testShouldStopPartitionCountMonitorWhenStoppingTheJobCoordinator() {
   }
 
   @Test
-  public void testStartpointFanOutOnBecomingLeader() throws IOException {
+  public void testLoadMetadataResources() throws IOException {
     ZkKeyBuilder keyBuilder = Mockito.mock(ZkKeyBuilder.class);
     ZkClient mockZkClient = Mockito.mock(ZkClient.class);
     when(keyBuilder.getJobModelVersionBarrierPrefix()).thenReturn(TEST_BARRIER_ROOT);
 
     ZkUtils zkUtils = Mockito.mock(ZkUtils.class);
     when(zkUtils.getKeyBuilder()).thenReturn(keyBuilder);
     when(zkUtils.getZkClient()).thenReturn(mockZkClient);
-    when(zkUtils.getJobModel(TEST_JOB_MODEL_VERSION)).thenReturn(new JobModel(new MapConfig(), new HashMap<>()));
-
-    ScheduleAfterDebounceTime mockDebounceTimer = Mockito.mock(ScheduleAfterDebounceTime.class);
-
-    Map<String, String> configMap = ImmutableMap.of(
-        "job.coordinator.system", "kafka",
-        "job.name", "test-job",
-        "systems.kafka.samza.factory", "org.apache.samza.system.MockSystemFactory");
-    MapConfig config = new MapConfig(configMap);
+    when(zkUtils.getJobModel(TEST_JOB_MODEL_VERSION)).thenReturn(jobModel);
 
     StartpointManager mockStartpointManager = Mockito.mock(StartpointManager.class);
     ZkJobCoordinator zkJobCoordinator = Mockito.spy(new ZkJobCoordinator("TEST_PROCESSOR_ID", config, new NoOpMetricsRegistry(), zkUtils));
     doReturn(mockStartpointManager).when(zkJobCoordinator).createStartpointManager(any(CoordinatorStreamStore.class));
     doReturn(mock(CoordinatorStreamStore.class)).when(zkJobCoordinator).createCoordinatorStreamStore();
-
-    StreamPartitionCountMonitor monitor = Mockito.mock(StreamPartitionCountMonitor.class);
-    zkJobCoordinator.debounceTimer = mockDebounceTimer;
-    zkJobCoordinator.streamPartitionCountMonitor = monitor;
-    when(zkJobCoordinator.getPartitionCountMonitor()).thenReturn(monitor);
-
-    Set<SystemStreamPartition> ssps = ImmutableSet.of(
-        new SystemStreamPartition("system1", "stream1_1", new Partition(0)),
-        new SystemStreamPartition("system1", "stream1_2", new Partition(0)));
-    Map<TaskName, TaskModel> tasksForContainer = ImmutableMap.of(
-        new TaskName("t1"), new TaskModel(new TaskName("t1"), ssps, new Partition(0)));
-    ContainerModel containerModel = new ContainerModel("0", tasksForContainer);
-    doReturn(new JobModel(config, ImmutableMap.of("0", containerModel))).when(zkJobCoordinator).generateNewJobModel(any());
+    doReturn(mock(MetadataResourceUtil.class)).when(zkJobCoordinator).createMetadataResourceUtil(any(JobModel.class));
```

Comment:
Maybe also verify `createResources` gets called on the `MetadataResourceUtil`?

Reply:
Will do.